dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/App.java

```diff
@@ -34,9 +34,39 @@ public static int[] mergeSort(int[] array){
     public static int[] mergeSortHelper(int[] arr, int start, int end){
         int[] newArray = new int[end - start + 1]; 
         int length = end - start + 1;
+       
+        if (end - start > 1){
+            int leftEnd = start + length / 2;
+            int rightStart = leftEnd + 1;
+            mergeSortHelper(arr, start, leftEnd);       // left half
+            mergeSortHelper(arr, rightStart, end);     // right half
+            
+            for (int i = start; i < rightStart; i++){
+                newArray[i - start] = arr[i];
+            }
 
-        mergeSortHelper(arr, start, start + (length / 2));       // left half
-        mergeSortHelper(arr, start + (length / 2), end + 1);     // right half
+            for (int i = rightStart; i <= end; i++){
+                for(int j = 0; j < newArray.length; j++){
+                    if (newArray[j] > arr[i]){
+                        for (int k = arr.length - 2; k >= j; k--){
+                            newArray[k + 1] = k;
+                        }
+                        newArray[j] = arr[i];
+                    }
+                }
+            }
+        }
+        else{ //Base case
+            //Check if arr requested part of the array is size 1 
+            if (start == end){
+                return arr;
+            }
+            int temp = arr[start];
+            if (arr[start] <= arr[end]){
+                arr[start] = arr[end];
+                arr[end] = temp;
+            }
+        }
         //Split into left and right
         //Recursively call mergeSortHelper on left and right halves
 
```

Comment:
It seems to be an incomplete part of the code which swaps the start and end of the arr array then places the end in a temporary variable.